packages/main/src/Table.js

```diff
@@ -446,6 +491,82 @@ class Table extends UI5Element {
 		this.fireEvent("load-more");
 	}
 
+	_handleSingleSelect(event) {
+		const row = this.getRowParent(event.target);
+		if (!row.selected) {
+			const previouslySelectedRows = this.rows.filter(item => item.selected);
+			this.rows.forEach(item => {
+				if (item.selected) {
+					item.selected = false;
+				}
+			});
+			row.selected = true;
+			this.fireEvent("selection-change", {
+				selectedRows: [row],
+				previouslySelectedRows,
+			});
+		}
+	}
+
+	_handleMultiSelect(event) {
+		const row = this.getRowParent(event.target);
+		const previouslySelectedRows = this.rows.filter(item => item.selected);
+
+		row.selected = !row.selected;
+
+		const selectedRows = this.rows.filter(item => item.selected);
```

Comment:
this.rows.filter(item => item.selected) is something you need in several places, add a getter for that:

```js
get selectedRows() {
    return this.rows.filter(item => item.selected)
}
```